scipy/sparse/base.py

```diff
@@ -249,7 +249,7 @@ def dot(self, other):
         array([ 1, -3, -1], dtype=int64)
 
         """
-        return self * other
+        return self.__mul__(other)
```

Comment:
The difference between the two is that `__mul__`, `__matmul__` correspond to Python's binop override mechanism, where `NotImplemented` as a return value has a special meaning, and will automatically result to TypeError if both sides punt.

`dot` on the other hand is an ordinary method, not in the binop system, which is why it should dispatch to `*`.